src/validation/coagulation/coag_1subarea.cpp

```diff
@@ -0,0 +1,112 @@
+#include <mam4xx/coagulation.hpp>
+#include <skywalker.hpp>
+#include <validation.hpp>
+
+using namespace skywalker;
+using namespace mam4;
+
+void coag_1subarea(Ensemble *ensemble) {
+
+  // Run the ensemble.
+  ensemble->process([=](const Input &input, Output &output) {
+    if (!input.has_array("deltat")) {
+      std::cerr << "Required name: "
+                << "deltat" << std::endl;
+      exit(1);
+    }
+    if (!input.has_array("pmid")) {
+      std::cerr << "Required name: "
+                << "pmid" << std::endl;
+      exit(1);
+    }
+    if (!input.has_array("aircon")) {
+      std::cerr << "Required name: "
+                << "aircon" << std::endl;
+      exit(1);
+    }
+    if (!input.has_array("dgn_a")) {
+      std::cerr << "Required name: "
+                << "dgn_a" << std::endl;
+      exit(1);
+    }
+    if (!input.has_array("dgn_awet")) {
+      std::cerr << "Required name: "
+                << "dgn_awet" << std::endl;
+      exit(1);
+    }
+    if (!input.has_array("wetdens")) {
+      std::cerr << "Required name: "
+                << "wetdens" << std::endl;
+      exit(1);
+    }
+    if (!input.has_array("qnum_cur")) {
+      std::cerr << "Required name: "
+                << "qnum_cur" << std::endl;
+      exit(1);
+    }
+    if (!input.has_array("qaer_cur")) {
+      std::cerr << "Required name: "
+                << "qaer_cur" << std::endl;
+      exit(1);
+    }
+
+    // read input data
+    auto deltat_f = input.get_array("deltat");
+    auto temp_f = input.get_array("temp");
+    auto pmid_f = input.get_array("pmid");
+    auto aircon_f = input.get_array("aircon");
+    auto dgn_a_f = input.get_array("dgn_a");
+    auto dgn_awet_f = input.get_array("dgn_awet");
+    auto wetdens_f = input.get_array("wetdens");
+    auto qnum_cur_f = input.get_array("qnum_cur");
+    auto qaer_cur_f = input.get_array("qaer_cur");
+
+    const int num_modes = AeroConfig::num_modes();
+    const int num_aero = AeroConfig::num_aerosol_ids();
+    const int max_agepair = Coagulation::max_agepair;
+    Real qaer_cur_c[num_aero][num_modes];
+    int n = 0;
+    for (int imode = 0; imode < num_modes; ++imode) {
+      for (int ispec = 0; ispec < num_aero; ++ispec) {
+        qaer_cur_c[ispec][imode] = qaer_cur_f[n];
+        n += 1;
+      }
+    }
+
+    Real qaer_del_coag_out_c[num_aero][max_agepair];
+    for (int imode = 0; imode < num_modes; ++imode) {
```

Comment:
You can replace this code with:
`Real qaer_del_coag_out_c[num_aero][max_agepair] = {0}`